### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by Malabarba**
Most important change is that doc buffer splits arglists into several lines.

Instead of this:

    org.httpkit.client/post
    ([url & [opts callback]] [url & [callback]])
      Issues an async HTTP POST request. See `request` for details.

We get this:

    org.httpkit.client/post
     [url & [opts callback]]
     [url & [callback]]
      Issues an async HTTP POST request. See `request` for details.

Which I find much easier to read.

### CODE REVIEW

File: cider-doc.el [hunk 1]
```diff
@@ -402,7 +402,7 @@ Tables are marked to be ignored by line wrap."
         (when (or forms args)
           (emit (cider-font-lock-as-clojure (or forms args))))
         (when (or special macro)
-          (emit (if special "Special Form" "Macro") 'font-lock-comment-face))
+          (emit (if special "Special Form" "Macro") 'font-lock-variable-name-face))
```
Comment (bbatsov) [inline cider-doc.el]: I'm guessing this is dependant on the theme.

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ an individual test using `C-c C-t t`.
 
 ### Changes
 
+* Doc buffer splits arglists into several lines
```
Comment (bbatsov) [inline CHANGELOG.md]: Add a `.` here for consistency with the other entries.

**Comment by bbatsov** [comment]
I like the change.

File: cider-doc.el (hunk 1 above)
Comment (Malabarba) [inline cider-doc.el]: Yes, but the point was just to change it to something other than `comment`. Comment face is pretty hidden on a lot of themes.

In the end it's personal preference. I can revert if desired.

**Pull request merged**